src/hotspot/share/cds/filemap.cpp

```diff
@@ -842,6 +864,50 @@ bool FileMapInfo::classpath_failure(const char* msg, const char* name) {
   return false;
 }
 
+unsigned int FileMapInfo::longest_common_app_classpath_len(int num_paths,
+                                                  GrowableArray<const char*>* rp_array) {
+  if (num_paths == 0) {
+    return 0;
+  }
+  unsigned int pos;
+  for (pos = 0; ; pos++) {
+    for (int i = 0; i < num_paths; i++) {
+      if (rp_array->at(i)[pos] != '\0' && rp_array->at(i)[pos] == rp_array->at(0)[pos]) {
+        continue;
+      }
+
+      // search backward for the pos before the file separator char
+      while (pos > 0 && rp_array->at(0)[--pos] != *os::file_separator());
+      return pos;
+    }
+  }
+  return 0;
+}
+
+bool FileMapInfo::check_paths_ignoring_common_path(int shared_path_start_idx, int num_paths,
```

Comment:
I think we should use the word "prefix" in the names. This way, it's clear that we are talking about a portion of a path string. How about `check_paths_ignoring_common_prefix`?

Reply:
Done.